src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java

```diff
@@ -107,6 +128,181 @@ public Person getSystemUser() {
+        //Returning true for null here is valid. 
+        //"Country of Incorporation/Citizenship" is not a required field.
+        //That value is left blank on the edoc for US vendors when their data 
+        //is either hand keyed into KFS or entered through Purchasing's AVF form.
+        return (ObjectUtils.isNull(vendorDetail.getVendorHeader().getVendorCorpCitizenCode()) 
+                || StringUtils.equalsIgnoreCase(vendorDetail.getVendorHeader().getVendorCorpCitizenCode(), KFSConstants.COUNTRY_CODE_UNITED_STATES));
+    }
+
+    private PaymentWorksVendor populateKfsEnteredPaymentWorksVendor(String kfsVendorDocumentNumber, VendorDetail vendorDetail, String kfsVendorProcessingStatus) {
+        PaymentWorksVendor pmwVendor = new PaymentWorksVendor();
+        pmwVendor.setPmwTransactionType(PaymentWorksConstants.PaymentWorksTransactionType.KFS_ORIGINATING_VENDOR);
+        pmwVendor.setKfsVendorProcessingStatus(kfsVendorProcessingStatus);
+        pmwVendor.setKfsVendorDocumentNumber(kfsVendorDocumentNumber);
+        pmwVendor.setProcessTimestamp(getDateTimeService().getCurrentTimestamp());
+        
+        pmwVendor.setKfsVendorHeaderGeneratedIdentifier(vendorDetail.getVendorHeaderGeneratedIdentifier());
+        pmwVendor.setKfsVendorDetailAssignedIdentifier(vendorDetail.getVendorDetailAssignedIdentifier());
+        pmwVendor.setRequestingCompanyTin(vendorDetail.getVendorHeader().getVendorTaxNumber());
+        populateVendorLegalName(pmwVendor, vendorDetail);
+        populateRemitAddress(pmwVendor, vendorDetail);
+        populateContactEmail(pmwVendor, vendorDetail);
+        return pmwVendor;
+    }
+    
+    private void populateVendorLegalName(PaymentWorksVendor pmwVendor, VendorDetail vendorDetail) {
+        if (vendorNameIsBlank(vendorDetail)) {
+            pmwVendor.setRequestingCompanyLegalName(formatVendorName(vendorDetail));
+        }
+        else {
+            pmwVendor.setRequestingCompanyLegalName(vendorDetail.getVendorName());
+        }
+    }
+
+    // Concatenates the vendorLastName and a delimiter and the vendorFirstName fields in a similar manner to the
+    // KualiCo base code PRIVATE method org.kuali.kfs.vnd.document.VendorMaintainableImpl.setVendorName which 
+    // formats the business object name values just prior to saving the data to the database.
+    private String formatVendorName(VendorDetail vendorDetail) {
+        StringBuilder sb = new StringBuilder(vendorDetail.getVendorLastName() + VendorConstants.NAME_DELIM + vendorDetail.getVendorFirstName());
+        return sb.toString();
+    }
+    
+    private boolean vendorNameIsBlank(VendorDetail vendorDetail) {
+        return (StringUtils.isBlank(vendorDetail.getVendorName()));
+    }
+
+    private void populateRemitAddress(PaymentWorksVendor pmwVendor, VendorDetail vendorDetail) {
+        List<VendorAddress> remitAddresses = findAllActiveRemitAddresses(vendorDetail.getVendorAddresses());
+        if (remitAddresses.size() == 1) {
```

Comment:
do we need to check for nulls here or does this always return a list?

Reply:
Adding as suggested